framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -993,32 +991,48 @@ private void checkContractsAtMethodDeclaration(
         if (contracts.isEmpty()) {
             return;
         }
-
-        JavaExpressionContext jeContext =
-                JavaExpressionContext.buildContextForMethodDeclaration(methodTree, checker);
-
+        StringToJavaExpression stringToJavaExpr =
+                stringExpr -> StringToJavaExpression.atMethodBody(stringExpr, methodTree, checker);
         for (Contract contract : contracts) {
             String expressionString = contract.expressionString;
             AnnotationMirror annotation = contract.annotation;
-
             annotation =
                     atypeFactory
                             .getDependentTypesHelper()
-                            .viewpointAdaptQualifierFromContract(annotation, jeContext, methodTree);
+                            .viewpointAdaptQualifierFromContract(
+                                    annotation, stringToJavaExpr, methodTree);
 
             JavaExpression exprJe;
             try {
-                exprJe = JavaExpressionParseUtil.parse(expressionString, jeContext);
+                // Parse the expressionString that was written on methodElement at the methodDecl.
```

Comment:
What is variable `methodDecl`?  Should this be "at the method declaration"?
Does "at" refer to "parse" or to "on methodElement"?  Please clarify.